grammars/gfm.cson

```diff
@@ -284,11 +284,11 @@
     'name': 'comment.hr.gfm'
   }
   {
-    'begin': '^\\s*[`~]{3,}\\s*(?i:(coffee-?(script)?))\\s*$'
+    'begin': '^\\s*(`{3,}|~{3,})\\s*(?i:(coffee-?(script)?|cson))\\s*$'
     'beginCaptures':
       '0':
         'name': 'support.gfm'
-    'end': '^\\s*[`~]{3,}$'
+    'end': '^\\s*\\1((?<=`)`+|(?<=~)~+)?\\s*$'
```

Comment:
Your proposed alternative is slower because it's unconditionally matching when there aren't any repeated characters. The tradeoff is exactly one character's difference in length:

~~~diff
-^\\s*\\1((?<=`)`+|(?<=~)~+)?\\s*$
+^\\s*\\1((?<=`)`*|(?<=~)~*)\\s*$
~~~